src/coreclr/nativeaot/CMakeLists.txt

```diff
@@ -30,6 +30,8 @@ if(CLR_CMAKE_HOST_UNIX)
 
   add_compile_options(-fno-rtti)          # Native AOT runtime doesn't use RTTI
   add_compile_options(-fno-exceptions)    # Native AOT runtime doesn't use C++ exception handling
+  add_compile_options(-nostdlib)
+  add_compile_options(-fno-threadsafe-statics)
```

Comment:
Is this going to generate buggy code with race conditions? I think we need to refactor any statics in local scopes that may depend on C++ runtime instead of passing this option.

Reply:
Without this flag, it requires `___cxa_guard_acquire` and `__cxa_guard_release` both on linux and osx.

> `-fno-threadsafe-statics`
> Do not emit the extra code to use the routines specified in the C++ ABI for thread-safe initialization of local statics. You can use this option to reduce code size slightly in code that doesn't need to be thread-safe.

Reply:
It is not ok to depend on race conditions to be reliably caught by CI.

Reply:
The code that triggers this looks like this and it will have a subtle race condition with this command line option:
```
class TypeWithConstructor
{
public:
    TypeWithConstructor();
};

TypeWithConstructor* m()
{
    static TypeWithConstructor x;
    return &x;
}
```
It needs to be changed to something like this so that the static is initialized at load time and the delayed initialization is not required:
```
class TypeWithConstructor
{
public:
    TypeWithConstructor();
};

static TypeWithConstructor x;

TypeWithConstructor* m()
{
    return &x;
}
```